modules/services/window-managers/hyprland.nix

```
@@ -0,0 +1,113 @@
+{ config, lib, pkgs, ... }:
+let cfg = config.wayland.windowManager.hyprland;
+in {
+  meta.maintainers = [ lib.maintainers.fufexan ];
+
+  options.wayland.windowManager.hyprland = {
+    enable = lib.mkEnableOption (lib.mdDoc "Hyprland wayland compositor");
+
+    package = lib.mkPackageOptionMD pkgs "hyprland" { };
+
+    finalPackage = cfg.package.override {
+      enableXWayland = cfg.xwayland.enable;
+      hidpiXWayland = cfg.xwayland.hidpi;
+      nvidiaPatches = cfg.enableNvidiaPatches;
+    };
+
+    plugins = lib.mkOption {
+      type = with lib.types; listOf (either package path);
+      default = [ ];
+      description = lib.mdDoc ''
+        List of Hyprland plugins to use. Can either be packages or
+        absolute plugin paths.
+      '';
+    };
+
+    systemdIntegration = lib.mkOption {
+      type = lib.types.bool;
+      default = pkgs.stdenv.isLinux;
+      description = lib.mdDoc ''
+        Whether to enable {path}`hyprland-session.target` on
+        hyprland startup. This links to `graphical-session.target`.
+        Some important environment variables will be imported to systemd
+        and dbus user environment before reaching the target, including
+        - `DISPLAY`
+        - `HYPRLAND_INSTANCE_SIGNATURE`
+        - `WAYLAND_DISPLAY`
+        - `XDG_CURRENT_DESKTOP`
+      '';
+    };
+
+    xwayland = {
+      enable = lib.mkEnableOption (lib.mdDoc "XWayland") // { default = true; };
+      hidpi = lib.mkEnableOption null // {
+        description = lib.mdDoc ''
+          Enable HiDPI XWayland, based on [XWayland MR 733](https://gitlab.freedesktop.org/xorg/xserver/-/merge_requests/733).
+          See <https://wiki.hyprland.org/Nix/Options-Overrides/#xwayland-hidpi> for more info.
+        '';
+      };
+    };
+
+    enableNvidiaPatches = lib.mkEnableOption
+      (lib.mdDoc "patching wlroots for better Nvidia support.");
```

The default description of `mkEnableOption` already puts a period at the end.

```suggestion
      (lib.mdDoc "patching wlroots for better Nvidia support");
```